### PR TITLE
bug fixed in ca_bimap and spectral bimap

### PR DESCRIPTION
**Pull request by yanzhaobiomath**
modify the function to make it possible to handle with extra input features

### CODE REVIEW

File: R/biMAP.R [hunk 1]
```diff
@@ -78,27 +81,34 @@ run_biMAP <- function(obj,
...
     if(!is.null(features)){
-      features = as.vector(features)
       ix = features %in% rownames(caobj@U)
       if (sum(ix) < length(features)){
         warning('only ', sum(ix),' out of ', length(features),  ' features are found from the caobj.')
       }
-      selected_items = c(selected_items, features)
+      
+      selected_items = c(selected_items, features[ix])
+      selected_items = selected_items[!duplicated(selected_items)]
```
Comment (ClemensKohl) [inline R/biMAP.R:96]: If I understand it correctly, what you want would be 
`selected_items = unique(selected_items)`
but I guess it has the same endresult.

File: R/biMAP.R [hunk 2]
```diff
@@ -110,33 +120,27 @@ run_biMAP <- function(obj,
...
-    genec <- rownames(caobj@U)
-  } else {
-    cellc <- names(cell_clusters(obj))
-    genec <- names(gene_clusters(obj))
-  }
-  
   colnames(umap_coords) <- c("x", "y")
   umap_coords$name <- rownames(umap_coords)
+  print(length(cellc) + length(genec))
+  print(dim(umap_coords))
```
Comment (ClemensKohl) [inline R/biMAP.R:126]: Please remove the print statements

File: R/biMAP.R [hunk 3]
```diff
@@ -78,27 +81,34 @@ run_biMAP <- function(obj,
...
       if (sum(ix) < length(features)){
         warning('only ', sum(ix),' out of ', length(features),  ' features are found from the caobj.')
       }
-      selected_items = c(selected_items, features)
+      
+      selected_items = c(selected_items, features[ix])
+      selected_items = selected_items[!duplicated(selected_items)]
+      
+      genec = c(genec, features[ix])
+      genec = genec[!duplicated(genec)]
```
Comment (ClemensKohl) [inline R/biMAP.R:99]: If I understand it correctly, what you want would be 
`selected_items = unique(selected_items)`
but I guess it has the same endresult.
Same for line 99

**Review by ClemensKohl — CHANGES_REQUESTED** [review]
Just remove the print() statements please, the rest is just a (cosmetic) suggestion.

File: R/biMAP.R (hunk 2 above)
Comment (yanzhaobiomath) [reply R/biMAP.R]: clear

File: R/biMAP.R (hunk 3 above)
Comment (yanzhaobiomath) [reply R/biMAP.R]: done

**Review by ClemensKohl — APPROVED** [review]
Awesome, thanks a lot :)

**Pull request merged**